wremnants/theory_tools.py

```diff
@@ -558,4 +564,24 @@ def pdfBugfixMSHT20(df , tensorPDFName):
         f"auto& res = {tensorPDFName};"
         f"res(15) = {tensorPDFName}(0) - ({tensorPDFName}(15) - {tensorPDFName}(0));"
         "return res")
-        
+
+def pdfHERAPDF20(df , tensorPDFName, entriesMain, branchExt, entriesExt):
+    # extend HERA pdfs with extra variations which should be symmetrized
+    # remove the first from the VAR branch, equal to the nominal
+    start = 0
+    entries_branch = 14 # total entries in the nano branch
+    df = df.Define(f"{tensorPDFName}_ext", f"auto res = wrem::clip_tensor(wrem::vec_to_tensor_t<double, {entries_branch}>({branchExt}, {start}), theory_weight_truncate); res = nominal_weight/central_pdf_weight*res; return res;")
+    df = df.Define(
+        f"{tensorPDFName}_ext_sym",
+        (f"Eigen::TensorFixedSize<double, Eigen::Sizes<{entriesMain+entriesExt}>> res;"
+        "for(int i=0; i<res.size(); i++) {"
+        f"   if(i < {entriesMain}) res(i) = {tensorPDFName}(i);"
+        "    else {"
+        f"      int k = i-{entriesMain}; res(i) = (k%2 == 0) ? {tensorPDFName}_ext(k/2+1) : ({tensorPDFName}(0) - ({tensorPDFName}_ext((k-1)/2+1) - {tensorPDFName}(0)));" # +1 to remove the first index
```

Comment:
Also does ``` {tensorPDFName}(0) - ({tensorPDFName}(i) - {tensorPDFName}(0))``` work generically to mirror the variation? I think @davidwalter2 checked this when he originally implemented it

Reply:
What kind of symmetrization do we want to do? Additive or multiplicative? If Multiplicative I would not explicitly symmetrize but use "mirror=True" when adding the systematic.

Reply:
> Are you writing this as [original_vales_i ... original_value_n, symmetrized_value_i...symmetrized_value_n]? It should be [origin_value_i, symmetrized_value_i, origin_value_i+1, symmetrized_value_i+1, ... ] for consistency with the rest of the set.

It is the latter: k%2==0 is the original and k%2==1 symmetrized

Reply:
Well, I think it would be better to not mirror the extended entries here but rather in the setupCombine in the addSystematic with mirror=True. What is done here is a multiplicative mirroring which leads to an asymmetric variation in the fit. With mirror=True you will get a symmetric variation

Reply:
Yes I agree, I'll adapt it to handle the mirroring in setupCombine.